application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -355,4 +355,15 @@ public String getTagName() {
             return tagName;
         }
     }
+
+    long getOpByHelpThreadId(final long channelId) {
+
+        logger.info("Looking for thread-record using channel ID: {}", channelId);
+
+        return database.read(context -> context.selectFrom(HelpThreads.HELP_THREADS)
+            .where(HelpThreads.HELP_THREADS.CHANNEL_ID.eq(channelId))
+            .fetchOptional()
+            .orElseThrow()
+            .getAuthorId());
```

Comment:
1. instead of fetching the entire row and then picking the only column ur interested in, directly only select that column: `select(author id).from(help threads)...`
2. it is not safe to assume that the method always finds a result - even in normal situations. for example if the bot was down for a few minutes and hours and threads have been created in the meantime. the bot is supposed to be able to naturally handly this situation, so u must not throw an exception here. instead go for returning `OptionalLong`.
3. id rename it to `getAuthor...` instead of `getOp...` for consistency with the rest of the code.